realm/realm-library/src/main/java/io/realm/internal/fields/CachedFieldDescriptor.java

```diff
@@ -82,8 +82,8 @@ protected void compileFieldDescription(List<String> fields) {
             columnType = tableInfo.getColumnType(columnName);
             if (i < nFields - 1) {
                 verifyInternalColumnType(currentTable, columnName, columnType);
-                currentTable = tableInfo.getLinkedTable(columnName);
             }
+            currentTable = tableInfo.getLinkedTable(columnName);
```

Comment:
Will, for example, `query.equalTo("abc", "backlinkField")` still throw? I guess so?